### ISSUE TITLE
Ill-typed publishDiagnostics message for code containing a specific operator

### ISSUE THREAD

**Issue by truls** [REditorSupport/languageserver#453]
Hi,

I originally posted this as a emacs lsp-mode bug (emacs-lsp/lsp-mode/issues/2914) but after doing some digging it appears that the R languageserver actually emits ill-typed `publishDiagnostics` messages for source code containing the `%<-%` operator from the zeallot package. See the previously referenced bug for a complete R code example.

An example of the ill-typed message is included below. Note the `"NA"` values for the `character` parameter.

```
[Trace - 11:30:52 AM] Received notification 'textDocument/publishDiagnostics'.
Params: {
  "uri": "file:///home/truls/phd/faas-project/data-analysis/util.R",
  "diagnostics": [
// ...
    {
      "range": {
        "start": {
          "line": 53,
          "character": "NA"
        },
        "end": {
          "line": 53,
          "character": "NA"
        }
      },
      "severity": 2,
      "source": "object_usage_linter",
      "message": "no visible global function definition for ‘%<-%’"
    },
// ...
```

**Comment by renkun-ken** [comment]
It is caused by https://github.com/jimhester/lintr/issues/482.

Closing as there's nothing we can do at the moment.

**Comment by truls** [comment]
Thanks, but you are misunderstanding my request. This isn't about the erroneous "no visisble global function" message but rather that the language server generates JSON where the `"character"` field is set to the string `"NA"`rather than an integer. This violates the [LSP specification](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#range) and may compliant clients to misbehave as is the case with Emacs' `lsp-mode`.

**Comment by renkun-ken** [comment]
Sorry for the misunderstanding. Would you like to try the latest lintr?

**Comment by truls** [comment]
The issue isn't present with lintr from master so I guess this will be resolved when lintr 3.0 is released. Thanks for your help!
